var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py

```diff
@@ -21,13 +23,18 @@ class PySphinxRtdTheme(PythonPackage):
     version("0.5.0", sha256="22c795ba2832a169ca301cd0a083f7a434e09c538c70beb42782c073651b707d")
     version("0.4.3", sha256="728607e34d60456d736cc7991fd236afb828b21b82f956c5ea75f94c8414040a")
 
+    depends_on("python@3.8:", when="@3:")
+    depends_on("python@3.6:", when="@:2")
+    depends_on("python@:3.11", when="@:1.0.0")  # distutils was removed in python@3.12
     depends_on("py-setuptools", type="build")
-
-    depends_on("py-sphinx@1.6:6", when="@1:", type=("build", "run"))
+    depends_on("py-sphinx@6:9", when="@2:", type=("build", "run"))
```

Comment:
Version ranges are inclusive so I think you want something along the lines:

```suggestion
    depends_on("py-sphinx@5:7", when="@2", type=("build", "run"))
    depends_on("py-sphinx@6:8", when="@3:", type=("build", "run"))

```

See the docs in and around https://spack.readthedocs.io/en/latest/packaging_guide.html#ranges-versus-specific-versions.